src/daml/Daml/Trading/Account/Account.daml

```diff
@@ -138,6 +144,52 @@ template Account
         
         create this with optNftAccount = Some (accountFactoryCid, holdingFactoryCid, accountKey)
     
+    choice Redeem_Badge: ContractId Account
+      with  
+        redeem_badge: Badge.BadgeType
+      controller owner
+      do
+        contractBadgeList <- forA badges \badge -> fetch badge
+        let
+          badgeUserResult = filter (\contractBadge -> contractBadge.badge == redeem_badge) contractBadgeList
+          currentRewardPoints = rewardPoints
+        assertMsg "Exists Redeem Badge!" (null badgeUserResult)
```

Comment:
Change message to something more meaningful. Ex: "Trader already owns this badge"